vscode/src/testController.ts

```diff
@@ -400,4 +416,148 @@ export class TestController {
 
     return testItem;
   }
+
+  private async resolveHandler(
+    item: vscode.TestItem | undefined,
+  ): Promise<void> {
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    if (!workspaceFolders) {
+      return;
+    }
+
+    if (item) {
+      const workspaceFolder = vscode.workspace.getWorkspaceFolder(item.uri!)!;
+
+      // If the item is a workspace, then we need to gather all test files inside of it
+      if (item.tags.some((tag) => tag === WORKSPACE_TAG)) {
+        await this.gatherWorkspaceTests(workspaceFolder, item);
+      } else {
+        // TODO: we are resolving the children of a specific test file. Here we will ask the server to parse the file,
+        // run all available listeners and then populate all available groups and examples
+      }
+    } else if (workspaceFolders.length === 1) {
+      // If there's only one workspace, there's no point in nesting the tests under the workspace name
+      await this.gatherWorkspaceTests(workspaceFolders[0], undefined);
+    } else {
+      // If there's more than one workspace, we use them as the top level items
+      for (const workspaceFolder of workspaceFolders) {
+        // Check if there is at least one Ruby test file in the workspace, otherwise we don't consider it
+        const pattern = new vscode.RelativePattern(
+          workspaceFolder,
+          "**/{test,spec}/**/{*_test,test_*,*_spec}.rb",
+        );
+        const files = await vscode.workspace.findFiles(pattern, undefined, 1);
+        if (files.length === 0) {
+          continue;
+        }
+
+        const uri = workspaceFolder.uri;
+        const testItem = this.testController.createTestItem(
+          uri.toString(),
+          workspaceFolder.name,
+          uri,
+        );
+        testItem.canResolveChildren = true;
+        testItem.tags = [WORKSPACE_TAG, DEBUG_TAG];
+        this.testController.items.add(testItem);
+      }
+    }
+  }
+
+  private async gatherWorkspaceTests(
+    workspaceFolder: vscode.WorkspaceFolder,
+    item: vscode.TestItem | undefined,
+  ) {
+    const pattern = new vscode.RelativePattern(
+      workspaceFolder,
+      "**/{test,spec,features}/**/{*_test.rb,test_*.rb,*_spec.rb,*.feature}",
```

Comment:
Why is this pattern different then the one on 445? Should we create a helper function for this?

Reply:
That was a mistake. Created a function to avoid mismatches.